### PR TITLE
Update example with advertisement and discovery

### PR DESCRIPTION
**Pull request by justprosh**

### CODE REVIEW

File: example/README.md [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+
+    ```bash
+    fluence init .
+    ```
+
+3. You can use [VSCode with Aqua extension](https://marketplace.visualstudio.com/items?itemName=FluenceLabs.aqua) for syntax highlighting  and better developer experience.
+
+## Add echo service in rust
+
+0. If you have **Apple Silicon** you should install cargo manually:
```
Comment (ig0r) [inline example/README.md:38]: I'd add "... manually, or you can ignore this step otherwise:"

File: example/README.md [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+    ```bash
+    npm i -g @fluencelabs/cli@0.2.7
+    ```
+
+2. Initialize Fluence project:
+
+    ```bash
+    fluence init .
+    ```
+
```
Comment (ig0r) [inline example/README.md:33]: Probably makes sense to add the output:

```
Successfully initialized Fluence project template at <your-path-to-registry-repo>/example

```

File: example/README.md [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+### Deploy service and test it
+
+1. Deploy service:
+
+    ```bash
+    fluence deploy
+    ```
+
+    Output, peer ids and service ids may differ:
```
Comment (ig0r) [inline example/README.md:76]: I'd add into the beginning "Please note..."

File: example/README.md [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+    ```
+
+    output:
+
+    ```bash
+    $ fluence service add echoService
+    Added echoService to fluence.yaml
+    ```
+
+2. Following [code](echoService/modules/echoService/src/main.rs) returns peer id of the host and initial message:
```
Comment (ig0r) [inline example/README.md:58]: The following code returns a peer id of the host and an initial message:

File: example/README.md [hunk 5]
```diff
@@ -0,0 +1,342 @@
...
+
+    echoService:
+    default:
+        - blueprintId: fd21793bae96d1519599346591bfe7ea75614d9fe7f7c824b279e5a2dc841ee0
+        serviceId: 785487e4-de08-4fa4-bdfd-df981ec66da7
+        peerId: 12D3KooWCMr9mU894i8JXAFqpgoFtx6qnV1LFPSfVc3Y34N4h4LS
+    ```
+
+
+2.  The following [code](src/aqua/main.aqua) declares module, imports App and EchoService and defines method to call `echo`:
```
Comment (ig0r) [inline example/README.md:109]: ...declares a module..., ..., and defines a method to call `echo`:

File: example/README.md [hunk 6]
```diff
@@ -0,0 +1,342 @@
...
+    Running:
+      function: echo("hi")
+      relay: /dns4/kras-08.fluence.dev/tcp/19001/wss/p2p/12D3KooWFtf3rfCDAfWwt6oLZYZbDfn9Vn7bv7g6QjjQxUUEFVBt
+    ... done
+
+    Result:
+
+    "12D3KooWCMr9mU894i8JXAFqpgoFtx6qnV1LFPSfVc3Y34N4h4LS: hi"
+    ```
+
```
Comment (ig0r) [inline example/README.md:146]: A short summary for the first part of the example would be great. Something like: "We've successfully build and deployed a service written in Rust, and called it from our Aqua code using the `fluence` CLI"

File: example/README.md [hunk 7]
```diff
@@ -0,0 +1,342 @@
...
+2. Initialize Fluence project:
+
+    ```bash
+    fluence init .
+    ```
+
+3. You can use [VSCode with Aqua extension](https://marketplace.visualstudio.com/items?itemName=FluenceLabs.aqua) for syntax highlighting  and better developer experience.
+
+## Add echo service in rust
```
Comment (ig0r) [inline example/README.md:36]: in Rust

File: example/README.md [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+
+2. Check out [src/aqua/export.aqua](src/aqua/export.aqua) with an export of EchoService API to JS/TS:
+
+    ```bash
+    module Export
+    import "services/echoService.aqua"
+    export EchoService
+    ```
+
+3. The following [code](src/echo.ts) registers local EchoService:
```
Comment (ig0r) [inline example/README.md:164]: Makes sence to specify a line number in the link src/echo.ts#L23

File: example/README.md [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+
+    ```rust
+    func echoJS(peer: string, relay: string, serviceId: string, msg: string) -> string:
+        on peer via relay:
+            EchoService serviceId
+            res <- EchoService.echo(msg)
+        <- res
+    ```
+    <a id="running-service"></a>
+5. Start client with service: `npm run start`
```
Comment (ig0r) [inline example/README.md:184]: I'd change to: Start the client with our service

File: example/README.md [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+    ...
+    > example@1.0.0 start
+    > node src/echo.js
+
+    📗 created a fluence peer 12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN with relay 12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi
+    Copy this code:
+    fluence run -f 'echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")'
+    ```
+
+6. Open a new terminal and copy the last line and execute to check echo service:
```
Comment (ig0r) [inline example/README.md:195]: Open a new terminal in the same registry example directory, and execute...

File: example/README.md [hunk 11]
```diff
@@ -0,0 +1,342 @@
...
+    Running:
+      function: echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")
+      relay: /dns4/kras-09.fluence.dev/tcp/19001/wss/p2p/12D3KooWD7CvsYcpF9HE9CCV9aY3SJ317tkXVykjtZnht2EbzDPm
+    ... done
+
+    Result:
+
+    "12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN: msg"
+    ```
+
```
Comment (ig0r) [inline example/README.md:208]: Again, a short summary (a couple of sentences) for the JS/TS part of the example would be helpful.

File: example/README.md [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+
+## Register both services in Registry
+
+If we, as service providers, want to be all echo services discoverable without specifying exact peer and service ids, we should use Registry to **advertise** our services.
```
Comment (ig0r) [inline example/README.md:212]: I'd rephrase it: 

As service providers we would like all our echo services to be discoverable without specifying particular peer and service ids, in order to achieve that we should use Registry to **advertise** our services.

File: example/README.md [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+
+Firstly, we need to create a **resource.** A resource represents a group of service providers and has a corresponding resource id for its discovery. Secondly, we need to register service providers on this resource id. So echo services can be discovered and resolved by resource id only.
```
Comment (ig0r) [inline example/README.md:214]: ... Secondly, we need to register the service providers on this resource id. So the echo services can be discovered and resolved by the resource id only.

File: example/README.md [hunk 14]
```diff
@@ -0,0 +1,342 @@
...
+
+1. The following [code](src/aqua/main.aqua) registers resource with label `echo`:
```
Comment (ig0r) [inline example/README.md:216]: added an anchor - src/aqua/main.aqua#L24

File: example/README.md [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+    ... done
+
+    Result:
+
+    [
+      "echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77"
+    ]
+    ```
+
+    So the resource id is `echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77`. By this resource id we can access any registered provider of this resource. There could be more registered echo service on different peers that we can smoothly use.
```
Comment (ig0r) [inline example/README.md:246]: I'd rephrase it:

So the resource id is `echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77`. Please not that the resource id might be different in your case. 

Using the resource id we can access any registered provider of the resource. There could be more registered echo services on different peers that we can use transparently.

File: example/README.md [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+3. This [code](src/aqua/main.aqua) registers deployed service by given `resource_id`:
+
+    ```rust
+    func registerService(resource_id: string) -> bool:
+        services <- App.services()
+        echoService = services.echoService.default[0]
+        success, error <- registerNodeProvider(echoService.peerId, resource_id, "", ?[echoService.serviceId])
+        <- success
+    ```
+
```
Comment (ig0r) [inline example/README.md:257]: A short Aqua code explainer might be helpful here

File: example/README.md [hunk 17]
```diff
@@ -0,0 +1,342 @@
...
+    Running:
+      function: registerService("echo12D3KooWCWsJLrG6evQKPEsDmKNbaW8NGj5JhVZfHumSvyx1Zw4c")
+      relay: /dns4/kras-05.fluence.dev/tcp/19001/wss/p2p/12D3KooWCMr9mU894i8JXAFqpgoFtx6qnV1LFPSfVc3Y34N4h4LS
+    ... done
+
+    Result:
+
+    true
+    ```
+
```
Comment (ig0r) [inline example/README.md:277]: I had the result with square brackets:

```
Result:

[
  [
    true
  ]
]
```

File: example/README.md [hunk 18]
```diff
@@ -0,0 +1,342 @@
...
+5. Next, we need to register JS service. For that we have Aqua imports and exports in [src/aqua/export.aqua](src/aqua/export.aqua):
+
+    ```
+    import registerProvider from "@fluencelabs/registry/resources-api.aqua"**
```
Comment (ig0r) [inline example/README.md:281]: Where do these two stars come from?

File: example/README.md [hunk 19]
```diff
@@ -0,0 +1,342 @@
...
+    export registerProvider
+    ```
+
+6. In [src/echo.ts](src/echo.ts) we should pass resource id as cmd argument:
```
Comment (ig0r) [inline example/README.md:285]: src/echo.ts#L32

File: example/README.md [hunk 20]
```diff
@@ -0,0 +1,342 @@
...
+`npm run start echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77` (Note: resource_id is from [the step 2](#resource-id))
+output:
+
+    ```rust
+    ...
+    Copy this code to call this service:
+    fluence run -f 'echoJS("12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN", "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi", "echo", "msg")'
+    registration result:  true
+    ```
+
```
Comment (ig0r) [inline example/README.md:302]: A short summary for the registering both the services helps to remember things

File: example/README.md [hunk 21]
```diff
@@ -0,0 +1,342 @@
...
+
+## Call services with Registry (using resource_id)
+
+1. In [src/aqua/main.aqua](src/aqua/main.aqua) defined `echoAll` to resolve services and call sequentially:
```
Comment (ig0r) [inline example/README.md:306]: src/aqua/main.aqua#L35

File: example/README.md [hunk 22]
```diff
@@ -0,0 +1,342 @@
...
+
+    Result:
+
+    [
+        [
+            "12D3KooWFEwNWcHqi9rtsmDhsYcDbRUCDXH84RC4FW6UfsFWaoHi: hi",
+            "12D3KooWCmnhnGvKTqEXpVLzdrYu3TkQ3HcLyArGJpLPooJQ69dN: hi"
+        ]
+    ]
+    ```
```
Comment (ig0r) [inline example/README.md:342]: Summary for the whole example that reiterates what was done and why it's needed. Can emphasise what else can be done and how the example might be developed further to help in daily development.

File: example/package.json [hunk 23]
```diff
@@ -1,30 +1,23 @@
 {
   "name": "example",
   "version": "1.0.0",
-  "description": "An example of how to use Registry in TypeScript",
-  "main": "dist/example.js",
+  "description": "",
```
Comment (ig0r) [inline example/package.json:4]: Need to add a description

Comment (justprosh) [reply example/package.json:4]: I don't think so we don't publish this

Comment (ig0r) [reply example/package.json:4]: Additional info (e.g. description) always helps

File: example/README.md [hunk 24]
```diff
@@ -73,7 +77,7 @@ rust: rustc 1.63.0-nightly
     fluence deploy
```
Comment (ig0r) [inline example/README.md:77]: Another thought. What if we add an additional step that is missing now and shows local testing with REPL using `fluence service repl echoService`? It's good to show a developer a full workflow.

The step could go either as a last step to the previous section or just before deployment.

File: example/README.md [hunk 25]
```diff
@@ -161,7 +166,7 @@ rust: rustc 1.63.0-nightly
     export EchoService
     ```
 
-3. The following [code](src/echo.ts) registers local EchoService:
+3. The following [code](src/echo.ts#L23) registers A local EchoService:
```
Comment (ig0r) [inline example/README.md:169]: should be small a

File: example/README.md [hunk 26]
```diff
@@ -0,0 +1,342 @@
...
+
+    output:
+    ```
+    $ fluence run -f 'echoAll("echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77", "hi")'
+    Running:
+    function: echoAll("echo12D3KooWRgEgxP4qAyUR5jerwKE1rwTLZoAhJ3YwAhpeSMC5pi77", "hi")
+    relay: /dns4/kras-06.fluence.dev/tcp/19001/wss/p2p/12D3KooWDUszU2NeWyUVjCXhGEt1MoZrhvdmaQQwtZUriuGN1jTr
+    ... done
+
+    Result:
```
[the remark this answers is not in the document]
Comment (justprosh) [reply example/README.md:385]: I fixed the code already

File: example/src/aqua/main.aqua [hunk 27]
```diff
@@ -0,0 +1,43 @@
...
+    <- results
+
+func echoJS(peer: string, relay: string, serviceId: string, msg: string) -> string:
+    on peer via relay:
+        EchoService serviceId
+        res <- EchoService.echo(msg)
+    <- res
+
+func registerResource() -> ?string:
+    resource_id, error <- createResource("echo")
```
Comment (boneyard93501) [inline example/src/aqua/main.aqua:25]: don't hardcode params.

File: example/README.md (hunk 16 above)
Comment (ig0r) [reply example/README.md:303]: What every line of the code does. So, it's clear what's happening when you call the function
